pkg/chains/formats/slsa/internal/material/material.go

```diff
@@ -215,3 +229,55 @@ func RemoveDuplicateMaterials(mats []common.ProvenanceMaterial) ([]common.Proven
 	}
 	return out, nil
 }
+
+// AddMaterialsFromPipelineParamsAndResults extracts type hinted params and results and adds the url and digest to materials.
+func AddMaterialsFromPipelineParamsAndResults(ctx context.Context, pro *objects.PipelineRunObject, mats []common.ProvenanceMaterial) []common.ProvenanceMaterial {
```

Comment:
Similarly, can we directly mutate the param?
```suggestion
func AddMaterialsFromPipelineParamsAndResults(ctx context.Context, pro *objects.PipelineRunObject, mats []*common.ProvenanceMaterial) {
```